src/strongbox/core.clj

```diff
@@ -1038,8 +1062,25 @@
 
 (def check-for-updates check-for-updates-in-parallel)
 
+(defn-spec find-installed-addon (s/or :match map? :no-match nil?)
+  [addon map?]
+  (let [keyfn (juxt :source :source-id)
+        key (keyfn addon)
+        addon-list (get-state :installed-addon-list)]
+    (first (filter (comp #(= key %) keyfn) addon-list))))
+
+(defn-spec check-addon-for-updates (s/or :ok :addon/installed, :app-not-started nil?)
+  "downloads full details for all installed addons that can be found in summary list"
+  [addon :addon/installed]
```

Comment:
we need more than just installed, we need a catalogue match to check for updates